packages/hdwallet-keepkey/src/binance.ts

```diff
@@ -45,16 +45,22 @@ export async function binanceSignTx(
     );
     if (resp.message_type === Core.Events.FAILURE) throw resp;
 
+    const intRegex = /^(0|([1-9][0-9]*))$/;
+    const outputAmount = message.outputs[0].coins[0].amount.toString();
+    if (!intRegex.test(outputAmount)) throw new Error("output amount should be an integer");
+    const inputAmount = message.inputs[0].coins[0].amount.toString();
+    if (!intRegex.test(inputAmount)) throw new Error("input amount should be an integer");
+
     let coinOut = new BinanceTransferMsg.BinanceCoin();
-    coinOut.setAmount(message.outputs[0].coins[0].amount.toString());
+    coinOut.setAmount(outputAmount);
```

Comment:
If using my previous suggestion
```suggestion
    coinOut.setAmount(outputAmount.toString());
```